dvc/repo/get.py

```diff
@@ -1,22 +1,38 @@
 import logging
 import os
+import shutil
 
 import shortuuid
 
 from dvc.exceptions import GetDVCFileError
 from dvc.exceptions import NotDvcRepoError
 from dvc.exceptions import OutputNotFoundError
 from dvc.exceptions import UrlNotDvcRepoError
+from dvc.exceptions import PathOutsideRepoError
 from dvc.external_repo import external_repo
 from dvc.path_info import PathInfo
 from dvc.stage import Stage
 from dvc.state import StateNoop
 from dvc.utils import resolve_output
 from dvc.utils.fs import remove
+from dvc.utils.compat import FileNotFoundError
 
 logger = logging.getLogger(__name__)
 
 
+def _copy_git_file(repo, src, dst, repo_url):
+    src_full_path = os.path.join(repo.root_dir, src)
+    dst_full_path = os.path.abspath(dst)
+
+    try:
+        if os.path.isdir(src_full_path):
+            shutil.copytree(src_full_path, dst_full_path)
+        else:
+            shutil.copy2(src_full_path, dst_full_path)
+    except FileNotFoundError:
```

Comment:
We are not really rasing against anything, so how about we
```
if not os.path.exists(src_full_path):
    raise PathOutsideRepoError(src, repo_url)
```
before "if os.path.isdir()" instead of wrapping it in try&except, to make it more linear? Looks like `shutil.copy2(src_full_path, dst_full_path)` is the only line that could raise this exception, as isdir will return False on non-existing path. Seems like it would make it easier to grasp. I don't have a strong opinion here. What do you think?

Reply:
Like so?
```py
def _copy_git_file(repo, src, dst, repo_url):
    src_full_path = os.path.join(repo.root_dir, src)
    dst_full_path = os.path.abspath(dst)

    if os.path.isdir(src_full_path):
        shutil.copytree(src_full_path, dst_full_path)
        return

    try:
        shutil.copy2(src_full_path, dst_full_path)
    except FileNotFoundError:
        raise PathOutsideRepoError(src, repo_url)

```

Reply:
@danihodovic Well, that would do it for me too 😄 Thanks! 🙂